scipy/stats/tests/test_hypotests.py

```diff
@@ -1482,6 +1425,20 @@ def test_engineering_stat_handbook(self):
             assert_allclose(lower[i, j], conf.low[i, j], atol=1e-2)
             assert_allclose(upper[i, j], conf.high[i, j], atol=1e-2)
 
+    def test_rand_symm(self):
+        np.random.seed(1234)
+        data = np.random.rand(3, 100)
+        res = stats.tukey_hsd(*data)
+        conf = res.confidence_interval()
+        # the confidence intervals should be negated symmetric of each other
+        assert_allclose(conf.low, -conf.high.T)
+        # upper diagonals of statistic should be negative symmetric to lower
+        # diagonals
+        assert_allclose(np.tril(res.statistic), -np.triu(res.statistic).T)
+        # p-values should be symmetric and 1 when compared to itself
+        assert_allclose(res.pvalue, res.pvalue.T)
+        assert_allclose(np.ones(3), np.diagonal(res.pvalue))
```

Comment:
```suggestion
        assert_equal(np.diagonal(res.pvalue), 1)
```
The `np.testing.assert_` comparisons are asymmetric , and the reference value should go on the right.
Also, consider `assert_equal` in this case since we would really expect exactly 1. I suppose if this fails then `assert_allclose` is fine, but I try `assert_equal` when I expect it to work consistently.

Reply:
Assert equal works fine for for all of these actually.